kiwi/boot/image/dracut.py

```diff
@@ -58,6 +62,75 @@ def include_file(self, filename, install_media=False):
             self.included_files_install.append('--install')
             self.included_files_install.append(filename)
 
+    def include_module(self, module):
+        """
+        Include module to dracut boot image
+
+        :param string module: module to include
+        """
+        if module not in self.modules:
+            self.modules.append(module)
+            self.modules_args = [
+                '--add', ' {0} '.format(' '.join(self.modules))
+            ]
+
+    def omit_module(self, module):
+        """
+        Omit module to dracut boot image
+
+        :param string module: module to omit
+        """
+        if module not in self.omit_modules:
+            self.omit_modules.append(module)
+            self.omit_modules_args = [
+                '--omit', ' {0} '.format(' '.join(self.omit_modules))
+            ]
+
+    def clear_modules_config(self):
+        """
+        Clears the modules configuration
+        """
+        self.modules = []
+        self.modules_args = []
+        self.omit_modules = []
+        self.omit_modiles_args = []
+
+    def persistent_config(self, config=None, config_file=None):
+        """
+        Writes modules configuration into a dracut configuration file.
+
+        If modules are not provided it uses the self modules attributes.
```

Comment:
Please don't let us do that. One of the primers of object oriented code is: write code for a single responsibility. The task of the method is to write a config file from given parameter information. Trying to be clever and use different information if no module data is provided extends the responsibility of the method a lot and makes it complex. The number of if statements in a code block is always a good indicator

Additionally the method is never called with empty modules. Thus you wrote code that is currently only used in the test. This is also something that should be avoided. I suggest to make the config parameter mandatory:

```
def write_system_config_file(self, config_params, config_file=None):
    ...
```

Reply:
config dictionary is mandatory now